src/asynk/async_queue.rs

```diff
@@ -233,61 +187,67 @@ impl AsyncQueueable for AsyncQueueTest<'_> {
 
         let metadata = serde_json::to_value(task)?;
 
-        let task: Task = if self.duplicated_tasks {
-            AsyncQueue::<NoTls>::insert_task_query(transaction, metadata, &task.task_type()).await?
+        let task: Task = if !task.uniq() {
+            AsyncQueue::<NoTls>::insert_task_query(
+                transaction,
+                metadata,
+                &task.task_type(),
+                Utc::now(),
+            )
+            .await?
         } else {
             AsyncQueue::<NoTls>::insert_task_if_not_exist_query(
                 transaction,
                 metadata,
                 &task.task_type(),
+                Utc::now(),
             )
             .await?
         };
         Ok(task)
     }
 
-    async fn schedule_next_task(
-        &mut self,
-        periodic_task: PeriodicTask,
-    ) -> Result<PeriodicTask, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        let periodic_task =
-            AsyncQueue::<NoTls>::schedule_next_task_query(transaction, periodic_task).await?;
-
-        Ok(periodic_task)
-    }
-    async fn insert_periodic_task(
-        &mut self,
-        task: &dyn AsyncRunnable,
-        timestamp: DateTime<Utc>,
-        period: i64,
-    ) -> Result<PeriodicTask, AsyncQueueError> {
+    async fn schedule_task(&mut self, task: &dyn AsyncRunnable) -> Result<Task, AsyncQueueError> {
         let transaction = &mut self.transaction;
 
         let metadata = serde_json::to_value(task)?;
 
-        let periodic_task = AsyncQueue::<NoTls>::insert_periodic_task_query(
-            transaction,
-            metadata,
-            timestamp,
-            period,
-        )
-        .await?;
-
-        Ok(periodic_task)
-    }
-
-    async fn fetch_periodic_tasks(
-        &mut self,
-        error_margin: StdDuration,
-    ) -> Result<Option<Vec<PeriodicTask>>, AsyncQueueError> {
-        let transaction = &mut self.transaction;
+        let scheduled_at = match task.cron() {
+            Some(scheduled) => match scheduled {
+                CronPattern(cron_pattern) => {
+                    let schedule = match Schedule::from_str(&cron_pattern) {
+                        Ok(schedule) => Ok(schedule),
+                        Err(_) => Err(AsyncQueueError::CronNotValidError),
+                    }?;
+                    let mut iterator = schedule.upcoming(Utc);
+                    iterator.next().unwrap()
```

Comment:
let's not use unwrap

Reply:
[Comments adressed](https://github.com/ayrat555/fang/pull/71/commits/e32b0fe5b77adb1ff9ec8a9453335b37a00bfccc)